File: src/main/java/com/ggemo/bililivedanmakuoop/cmddata/GuardBuyData.java
package com.ggemo.bililivedanmakuoop.cmddata;

import com.alibaba.fastjson.JSON;
import com.alibaba.fastjson.JSONObject;
import com.alibaba.fastjson.annotation.JSONField;
import com.ggemo.bililivedanmakuoop.CmdData;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

@Getter
@Setter
@AllArgsConstructor
@ToString
public class GuardBuyData implements CmdData {
    @JSONField(name = "uid")
    long userId;

    @JSONField(name = "username")
    String userName;

    @JSONField(name = "guard_level")
    int guardLevel;

    int num;

    long price;

    /**
     * 名叫name, 结果类型是number...
     */
    @JSONField(name = "role_name")
    long roleName;

    @JSONField(name = "gift_name")
    String giftName;

    @JSONField(name = "start_time")
    long startTime;

    @JSONField(name = "end_time")
    long endTime;

    public static GuardBuyData fromJSON(String json) {
        return fromJSON(JSON.parseObject(json).getJSONObject("data"));
    }

    public static GuardBuyData fromJSON(JSONObject json) {
        return json.toJavaObject(GuardBuyData.class);
    }

    public String toJSONString() {
        return JSON.toJSONString(this);
    }

    public static GuardBuyData fromGgemoJSON(JSONObject json){
        return json.toJavaObject(GuardBuyData.class);
    }

    public static GuardBuyData fromGgemoJSON(String json){
        return JSON.parseObject(json, GuardBuyData.class);
    }
}

File: src/main/java/com/ggemo/va/bililivedanmakuoop/msgdata/GuardBuyData.kt
package com.ggemo.va.bililivedanmakuoop.msgdata

import com.alibaba.fastjson.annotation.JSONField
import com.alibaba.fastjson.JSON
import com.alibaba.fastjson.JSONObject

class GuardBuyData : MsgData() {
    @JSONField(name = "uid")
    var userId: Long? = 0
    var userName: String? = null
    var guardLevel: Number? = 0
    var num: Number? = 0
    var price: Long? = 0

    /**
     * 名叫name, 结果类型是number...
     */
    var roleName: Long? = 0
    var giftName: String? = null
    var startTime: Long? = 0
    var endTime: Long? = 0

    companion object {
        fun fromJSON(json: String?): GuardBuyData {
            return fromJSON(JSON.parseObject(json).getJSONObject("data"))
        }

        fun fromJSON(json: JSONObject): GuardBuyData {
            return json.toJavaObject(GuardBuyData::class.java)
        }
    }
}